worker/uniter/pebblepoller_test.go

```diff
@@ -130,3 +130,6 @@ func (c *fakePebbleClient) TriggerStart() {
 	c.err = nil
 	c.sysInfo.BootID = utils.MustNewUUID().String()
 }
+
+func (c *fakePebbleClient) CloseIdleConnections() {
```

Comment:
Can we add something to make sure this is called as expected in the defer

Reply:
Or alternatively: 1) create the client once outside the poll loop, or 2) add `DisableKeepAlive: true` to the Pebble client config. I'd probably prefer the latter as I don't 100% trust keepalive in a long-running context like this. :-)